source/fundamentals/connection/connection-options.txt

```diff
@@ -6,203 +6,290 @@ Connection Options
+       private key file. If you need both, you must concatenate the files.
 
-   * - **srvServiceName**
+   * - **tlsCertificateKeyFilePassword**
      - string
-     - ``mongodb``
-     - Specifies the SRV record service name to which the driver should
-       connect.
+     - ``null``
+     - Specifies the password to decrypt the client private key to be used
+       for TLS connections.
 
-   * - **socketTimeoutMS**
-     - integer
-     - ``360000``
-     - Specifies the number of milliseconds to wait before timeout on a TCP
-       socket.
+   * - **tlsDisableCertificateRevocationCheck**
+     - boolean
+     - ``false``
+     - Specifies whether the driver will check a certificate's
+       revocation status via CRLs or OCSP.
 
-   * - **tls**
+   * - **tlsDisableOCSPEndpointCheck**
      - boolean
      - ``false``
-     - Specifies whether to establish a Transport Layer Security (TLS)
-       connection with the instance. This is automatically set to ``true``
-       when using a DNS seedlist (SRV) in the connection string. You can
-       override this behavior by setting the value to ``false``.
+     - Specifies whether the driver will reach out to OCSP
+       endpoints if needed.
 
-   * - **validateOptions**
+   * - **tlsInsecure**
      - boolean
      - ``false``
-     - Specifies whether to error when the method parameters contain an
-       unknown or incorrect option. If ``false``, the driver produces warnings
-       only.
+     - Specifies to relax TLS constraints as much as possible, such as
+       allowing invalid certificates or hostname mismatches.
+
+   * - **w**
+     - non-negative integer or string
+     - ``null``
+     - Specifies the default write concern "w" field for the client.
 
    * - **waitQueueTimeoutMS**
-     - integer
+     - non-negative integer
      - ``0``
-     - Specifies the maximum amount of time in milliseconds that operation
-       execution can wait for a connection to become available.
+     - Specifies the amount of time spent attempting to check out a connection
+       from a server's connection pool before timing out.
 
-   * - **writeConcern**
-     - string or integer
+   * - **wTimeoutMS**
+     - non-negative integer
      - ``null``
-     - Specifies the write concern. For more information on values, see the
-       server documentation on the
-       :manual:`w Option </reference/write-concern/#w-option>`.
-
-For a complete list of options, see the `MongoClient
-<{+api+}/classes/MongoClient.html>`__ API reference page. 
+     - Specifies the default write concern "wtimeout" field for the client.
+
+   * - **zlibCompressionLevel**
+     - integer between -1 and 9 (inclusive)
+     - ``-1``
+     - Specifies the level of compression when using zlib to compress wire
+       protocol messages. ``-1`` signifies the default level, ``0`` signifies
+       no compression, ``1`` signifies the fastest speed, and ``9`` signifies
+       the best compression.
```

Comment:
Suggestion: May be good to link to compression page here https://docs-mongodbcom-staging.corp.mongodb.com/node/docsworker-xlarge/update-connection-opts/fundamentals/connection/network-compression/